### PR TITLE
Clear trusted devices when disabling 2FA

### PR DESCRIPTION
**Pull request by bytehead**
See #1309.

### CODE REVIEW

**Review by leofeyer — DISMISSED** [review]

**Comment by leofeyer** [comment]
What about the backup codes? Are they removed as well?

**Comment by bytehead** [comment]
Yes, see Line 195:
`$user->backupCodes = null;`

File: core-bundle/tests/Controller/FrontendModule/TwoFactorControllerTest.php [hunk 1]
```diff
@@ -179,6 +179,7 @@ public function testRedirectsAfterTwoFactorHasBeenDisabled(): void
         /** @var RedirectResponse $response */
         $response = $controller($request, $module, 'main', null, $page);
 
+        $this->assertSame('', $user->useTwoFactor);
```
Comment (leofeyer) [inline core-bundle/tests/Controller/FrontendModule/TwoFactorControllerTest.php:192]: Maybe we can also assert here that the backup codes are removed?

Comment (bytehead) [reply core-bundle/tests/Controller/FrontendModule/TwoFactorControllerTest.php:192]: Yep! Thinking about to check as well if the token version has increased.

Comment (bytehead) [reply core-bundle/tests/Controller/FrontendModule/TwoFactorControllerTest.php:192]: Added in 63ae617b0. Also a check, if `->clearTrustedDevices()` is called.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @bytehead

**Pull request merged**